engine/select_optimal_members.py

```diff
@@ -0,0 +1,305 @@
+import os
+from collections import Counter
+from datetime import datetime
+
+import click
+import numpy as np
+
+from engine.check import check_intersection, check_variable
+from engine.tolerance import tolerance
+from util.click_util import cli_help
+from util.dataframe_ops import compute_rel_diff_dataframe, parse_probtest_csv
+from util.log_handler import logger
+
+
+def select_members(
+    stats_file_name,
+    member_type,
+    min_member_num,
+    max_member_num,
+    total_member_num,
+    min_factor,
+    max_factor,
+):
+
+    members = [i for i in range(1, total_member_num + 1)]
+
+    # Iteratively change likelihood of members being selected
+    # to come to a solution faster
+    weights = np.array([1 / total_member_num] * total_member_num)
+
+    # Use fixed seed to make result reproducible
+    np.random.seed(total_member_num)
+
+    for f in range(
+        int(min_factor), int(max_factor) + 1, 5
+    ):  # Try with bigger factor if max_member_num is not enough
+        logger.info("Set factor to {}".format(f))
+        for mem_num in range(min_member_num, max_member_num + 1):
+            logger.info("Try with {} members.".format(mem_num))
+            max_passed = 1
+            vars = []
+            for i in range(1, 51):
```

Comment:
With your above suggestions, `iter` starts at `0`. However, I find it nicer when it is the actual number of the iteration. Right now it gives the output "Test 1,... Test 2...." etc. Also the calculations for stopping earlier is based on `iter` starting at `1`. Of course, I could adapt it there instead but this would be less nice I think.

Reply:
I think using starting index 1 in a zero based language makes the code more complicated as needed. And for iterations I am really used to have them starting at zero. For the output you can just add a +1 there and adjust the criterion.